main.go

```diff
@@ -624,12 +629,27 @@ corectl eval by "Region" // Returns the values for dimension "Region"`,
 			removeCmd.PersistentPreRun(removeCmd, args)
 		},
 		Run: func(ccmd *cobra.Command, args []string) {
-			if len(args) != 1 {
+			app := viper.GetString("app")
+
+			if len(args) != 1 && app == "" {
 				fmt.Println("Expected an identifier of the app to delete.")
 				ccmd.Usage()
 				os.Exit(1)
 			}
-			internal.DeleteApp(rootCtx, viper.GetString("engine"), args[0], viper.GetString("ttl"), headers)
+
+			if len(args) == 1 {
+				app = args[0]
+			}
+
+			var confirmed bool
+
+			if viper.GetString("suppress") != "true" {
+				confirmed = askForConfirmation(fmt.Sprintf("Do you really want to delete the app: %s?", app))
+			}
+
+			if confirmed || viper.GetString("suppress") == "true" {
```

Comment:
Maybe its easier if we let `askForConfirmation` handle the `suppress` flag? That way we don't need to implement this logic in all methods that would require a confirm dialog later on.